src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchMultiSourceETLTest.java

```diff
@@ -163,4 +166,48 @@ public void testBlackList() throws Exception {
     Assert.assertEquals(expectedSObjects, actualSObjects);
   }
 
+  @Test
+  public void testValidWhiteListSObjects() {
+     SalesforceMultiSourceConfig salesforceMultiSourceConfig = new SalesforceMultiSourceConfigBuilder()
+       .setConsumerKey(CONSUMER_KEY).setConsumerSecret(CONSUMER_SECRET).setUsername(USERNAME).setPassword(PASSWORD)
+       .setSecurityToken(SECURITY_TOKEN).setLoginUrl(LOGIN_URL).setWhiteList("Account,Contact").build();
+
+     MockFailureCollector collector = new MockFailureCollector();
+     salesforceMultiSourceConfig.validateSObjects(collector);
+     Assert.assertEquals(collector.getValidationFailures().size(), 0);
+  }
+
+  @Test
+  public void testValidBlackListSObjects() {
+    SalesforceMultiSourceConfig salesforceMultiSourceConfig = new SalesforceMultiSourceConfigBuilder()
+      .setConsumerKey(CONSUMER_KEY).setConsumerSecret(CONSUMER_SECRET).setUsername(USERNAME).setPassword(PASSWORD)
+      .setSecurityToken(SECURITY_TOKEN).setLoginUrl(LOGIN_URL).setBlackList("Contact").build();
+
+    MockFailureCollector collector = new MockFailureCollector();
+    salesforceMultiSourceConfig.validateSObjects(collector);
+    Assert.assertEquals(collector.getValidationFailures().size(), 0);
+  }
+
+  @Test
+  public void testInvalidWhiteListSObjects() {
+    SalesforceMultiSourceConfig salesforceMultiSourceConfig = new SalesforceMultiSourceConfigBuilder()
+      .setConsumerKey(CONSUMER_KEY).setConsumerSecret(CONSUMER_SECRET).setUsername(USERNAME).setPassword(PASSWORD)
+      .setSecurityToken(SECURITY_TOKEN).setLoginUrl(LOGIN_URL).setWhiteList("invalidObject1,invalidObject2").build();
+
+    MockFailureCollector collector = new MockFailureCollector();
+    salesforceMultiSourceConfig.validateSObjects(collector);
+    Assert.assertEquals(collector.getValidationFailures().size(), 1);
+  }
+
+  @Test
+  public void testInvalidBlackListSObjects() {
+
+    SalesforceMultiSourceConfig salesforceMultiSourceConfig = new SalesforceMultiSourceConfigBuilder()
+      .setConsumerKey(CONSUMER_KEY).setConsumerSecret(CONSUMER_SECRET).setUsername(USERNAME).setPassword(PASSWORD)
+      .setSecurityToken(SECURITY_TOKEN).setLoginUrl(LOGIN_URL).setBlackList("invalidObject1,invalidObject2").build();
+
+    MockFailureCollector collector = new MockFailureCollector();
+    salesforceMultiSourceConfig.validateSObjects(collector);
+    Assert.assertEquals(collector.getValidationFailures().size(),  1);
```

Comment:
Is the number of validation failures one even if there are two invalid objects?

Reply:
yes the number of validation failures will be one.